Tests/LibWeb/TestBorderRadiusNormalization.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2025, Dan Vittegleo <dan@vittegleo.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibTest/TestCase.h>
+#include <LibWeb/Painting/BorderRadiiData.h>
+#include <LibWeb/PixelUnits.h>
+
+namespace Web {
+
+static auto calculate_scaling_factor(Painting::BorderRadiiData const& radii, CSSPixels width, CSSPixels height)
+{
+    auto top_sum = radii.top_left.horizontal_radius + radii.top_right.horizontal_radius;
+    auto bottom_sum = radii.bottom_left.horizontal_radius + radii.bottom_right.horizontal_radius;
+    auto left_sum = radii.top_left.vertical_radius + radii.bottom_left.vertical_radius;
+    auto right_sum = radii.top_right.vertical_radius + radii.bottom_right.vertical_radius;
+
+    auto f = CSSPixels(1.0);
+    if (top_sum > width)
+        f = min(f, width / top_sum);
+    if (bottom_sum > width)
+        f = min(f, width / bottom_sum);
+    if (left_sum > height)
+        f = min(f, height / left_sum);
+    if (right_sum > height)
+        f = min(f, height / right_sum);
+    return f;
+}
+
+TEST_CASE(border_radius_normalization_no_overlap)
+{
+    Painting::BorderRadiiData radii;
+    radii.top_left = radii.top_right = radii.bottom_right = radii.bottom_left = { CSSPixels(10), CSSPixels(10) };
+
+    auto f = calculate_scaling_factor(radii, CSSPixels(100), CSSPixels(100));
```

Comment:
you are testing the test; it’d be better to import the extracted snippet actually used in the application instead.

Reply:
moved this logic to `normalize_overlapping_radii` so it could be used directly in the tests